hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -81,4 +82,53 @@ object HoodieWriterUtils {
     params.foreach(kv => props.setProperty(kv._1, kv._2))
     props
   }
+
+  /**
+   * Get the partition columns to stored to hoodie.properties.
+   * @param parameters
+   * @return
+   */
+  def getPartitionColumns(parameters: Map[String, String]): Option[String] = {
+    val  keyGenClass = parameters.getOrElse(KEYGENERATOR_CLASS_OPT_KEY,
+      DEFAULT_KEYGENERATOR_CLASS_OPT_VAL)
+    try {
+      val constructor = getClass.getClassLoader.loadClass(keyGenClass)
+        .getConstructor(classOf[TypedProperties])
+      constructor.setAccessible(true)
+      val props = new TypedProperties()
+      props.putAll(parameters.asJava)
+      val keyGen = constructor.newInstance(props)
```

Comment:
Can't we pass the KeyGenerator already created in `HoodieSparkSqlWriter` and `DeltaSync` instead of recreating it again here ? Both these places already create `KeyGenerator` using reflection.

Reply:
Hi @umehrot2 , for bootstrap in `HoodieSparkSqlWriter`, there is no KeyGenerator created, so we need to recreating it here.

Reply:
Reuse the KeyGenerator except the bootstrap method.